sdk/rust/examples/backend/src/main.rs

```diff
@@ -0,0 +1,46 @@
+mod configuration;
+
+use eyre::Result;
+use dagger_sdk::{Query, File, Container};
+use clap::Parser;
+
+use configuration::Configuration;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let client = dagger_sdk::connect().await?;
+    let build = build_backend(&client).await;
+    let image = build_prod_image(&client, build).await;
+    let image_reference = push_image(image).await?;
+    println!("Image published at: {}", image_reference);
+    Ok(())
+}
+
+async fn build_backend(client: &Query) -> File {
+    let backend_directory = client.host().directory("axum-backend");
+    client
+        .container()
+        .from("rust:alpine")
+        .with_directory("./backend", backend_directory)
+        .with_workdir("/backend")
+        .with_exec(vec!["apk", "add", "build-base", "musl"])
+        .with_exec(vec!["cargo", "build", "--release"])
+        .file("./target/release/axum-backend")
+}
+
+async fn build_prod_image(client: &Query, build: File) -> Container {
+    let Configuration { port } = Configuration::parse();
+    client
+        .container()
+        .from("gcr.io/distroless/static-debian12")
+        .with_file(".", build)
+        .with_env_variable("PORT", port.to_string())
```

Comment:
we likely shouldn't set this here unless you of course want a default for the app. Otherwise this will only be set for the build and not when it is actually running.